samples/pmic/npmx/buck/src/main.c

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <npmx_buck.h>
+#include <npmx_driver.h>
+#include <npmx_gpio.h>
+
+#define LOG_MODULE_NAME pmic_buck
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Values used to refer to the specified buck. */
+#define BUCK1_IDX 0
+#define BUCK2_IDX 1
+
+#if CONFIG_TESTCASE_SET_BUCK_VOLTAGE || CONFIG_TESTCASE_OUTPUT_VOLTAGE ||                          \
+	CONFIG_TESTCASE_ENABLE_BUCKS_USING_PIN
+/**
+ * @brief Function for setting the buck output voltage for the specified buck instance.
+ *
+ * @param[in] p_buck  Pointer to the instance of buck.
+ * @param[in] voltage Selected voltage.
+ *
+ * @retval NPMX_SUCCESS  Operation performed successfully.
+ * @retval NPMX_ERROR_IO Error using IO bus line.
+ */
+static npmx_error_t set_buck_voltage(npmx_buck_t *p_buck, npmx_buck_voltage_t voltage)
```

Comment:
this functionality is available via upstream regulator API. Why do we need such sample in NCS? This is just confusing.

Reply:
Why isn't then this part of npmx repository? It needs to be clear that npmx is not the way to go if one expects good integration with NCS, but for bare-metal or custom stuff.